File: src/com/designpatterns/creational/builder/Chef.java
package com.designpatterns.creational.builder;

public class Chef implements IChef {

    @Override
    public Pizza getPepperoniPizza() {
        return new PizzaBuilder()
                .setCrustType(Crust.CheeseCrust)
                .setPepperoni(true)
                .setCapsicum(true)
                .setMozzarella(true)
                .setMushroom(false)
                .setOlives(false)
                .setSize(8)
                .build();
    }

    @Override
    public Pizza getSupremeVegPizza() {
        return new PizzaBuilder()
                .setCrustType(Crust.CheeseCrust)
                .setPepperoni(false)
                .setCapsicum(false)
                .setMozzarella(true)
                .setOlives(true)
                .setMushroom(true)
                .setSize(8)
                .build();
    }
}

File: src/com/designpatterns/creational/builder/kotlin/Chef.kt
package com.designpatterns.creational.builder.kotlin

class Chef : IChef {

    override val pepperoniPizza: Pizza
        get() = Pizza().apply {
            crustType = Crust.CheeseCrust
            isNeedPepperoni = true
            isNeedCapsicum = true
            isNeedMushroom = true
            size = 8
        }

    override val supremeVegPizza: Pizza
        get() = Pizza().apply {
            crustType = Crust.CheeseCrust
            isNeedMozzarella = true
            isNeedOlives = true
            isNeedMushroom = true
            size = 8
        }
}